pkg/volume/emptydir/empty_dir_test.go

```diff
@@ -963,27 +961,7 @@ func TestCalculateEmptyDirMemorySize(t *testing.T) {
 		nodeAllocatableMemory resource.Quantity
 		emptyDirSizeLimit     resource.Quantity
 		expectedResult        resource.Quantity
-		featureGateEnabled    bool
```

Comment:
@Jefftree how do we feel about removing gated tests at GA?  Even though the gat is "locked", there is still code under the gate and ultimately it's just a variable, right? Should we keep the tests until N+3?

Reply:
I think we have to remove it. If a test tries to change a feature gate I think it is a panic for GA flags.

Reply:
yeah, it may indicate that we either need to rewrite this test to enable compatibility mode instead. But we need to see what kind of compatibility guarantees we are documenting.

Reply:
You can keep the test by adding a line `featuregatetesting.SetFeatureGateEmulationVersionDuringTest(t, utilfeature.DefaultFeatureGate, version.MustParse("1.31"))` before setting the feature gate via `featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.SizeMemoryBackedVolumes, testCase.featureGateEnabled)`. An example can be seen here: https://github.com/kubernetes/kubernetes/pull/127348/files#diff-92f41de420a097c0419a3cb53cf176a459a742c39f3fa482ab48b1c2fa6fd084R6730.

The disablement test will be locked to emulation version 1.31 that k8s versions 1.32-1.34 will support.

Reply:
I don't think this works here. For E2E node, we set the feature gate on the kubelet config.

Reply:
sorry, this would be tricky for https://github.com/kubernetes/kubernetes/pull/126981/files#diff-587ea2030f2b92ebe8061f12bf42ae3027a51a9c43ff16ae8a77f7a9a1aa167c.

I think for the unit tests I can add this.

Reply:
deletion in e2e node should be fine because compatibility version is not supported in kubelet. The comment is referring to unit and integration tests.

Reply:
So should we be changing all unit tests that trigger features gates to use `featuregatetesting.SetFeatureGateEmulationVersionDuringTest(t, utilfeature.DefaultFeatureGate, version.MustParse("1.31"))`?

Or something like that?

Reply:
I'm not sure what is required from me for this and what we should enforce on all PRs for the next few weeks.

Reply:
Sorry I'm trying to understand this a bit better, is this used at all by control plane components or is it a kubelet only feature? Kubelet only features require no feature gate changes since compatibility version is not supported so if that's the case then this PR is lgtm for feature gates.

Keeping the tests described in https://github.com/kubernetes/kubernetes/pull/126981/files#r1801852232 is only necessary for control plane components. I see the volume package is imported by kcm though am having a bit of trouble understanding the full graph. 

Thanks!

cc @thockin @jpbetz

Reply:
The KEP states that Kubelet is the only component where this feature gate is used. Volumes confuse me a bit also but I think it is only kubelet. There isn't any API changes for this feature and it only changes some basic behavior with emptyDir.

Reply:
Okay, feature gate changes lgtm.